### DISCUSSION TITLE
Enable Explorer navigation pane expanding to current folder

### DISCUSSION THREAD

**Discussion by waldo2188** [farag2/Sophia-Script-for-Windows#444] [Ideas]
Hi,
In the old [Disassembler0](https://github.com/Disassembler0/Win10-Initial-Setup-Script) we have the ability to enabled/disabled 
_Explorer navigation pane expanding_. It's an option of Explorer.exe that you can enable through the [Explorer Options UI](https://www.elevenforum.com/t/turn-on-or-off-expand-to-current-folder-in-navigation-pane-of-file-explorer-in-windows-11.2544/) 
that allows the navigation pane automatically to expand to the current folder or drive as you open them in File Explorer.

It can be enabled by CMD like this :
`Set-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced" -Name "NavPaneExpandToCurrentFolder" -Type DWord -Value 1}`

And disabled like this :
`Remove-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced" -Name "NavPaneExpandToCurrentFolder" -ErrorAction SilentlyContinue`

This option can be put on the _UI & Personalization_ part.

**Comment by farag2** [comment]
Hm...

**Comment by waldo2188** [comment]
Is that a good "hm" or a bad one ?

**Comment by farag2** [comment]
```powershell
<#
	.SYNOPSIS
	Expand to current folder in navigation pane

	.PARAMETER Disable
	Do not expand to open folder on navigation pane (default value)

	.PARAMETER Enable
	Expand to open folder on navigation pane

	.EXAMPLE
	NavigationPaneExpand -Disable

	.EXAMPLE
	NavigationPaneExpand -Enable

	.NOTES
	Current user
#>
function NavigationPaneExpand ###
{
	param
	(
		[Parameter(
			Mandatory = $true,
			ParameterSetName = "Disable"
		)]
		[switch]
		$Disable,

		[Parameter(
			Mandatory = $true,
			ParameterSetName = "Enable"
		)]
		[switch]
		$Enable
	)

	switch ($PSCmdlet.ParameterSetName)
	{
		"Disable"
		{
			Remove-ItemProperty -Path HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced -Name NavPaneExpandToCurrentFolder -Force -ErrorAction Ignore
		}
		"Enable"
		{
			New-ItemProperty -Path HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced -Name NavPaneExpandToCurrentFolder -PropertyType DWord -Value 1 -Force
		}
	}
}
```

Soon it will be implemented with a new script release.

**Reply by farag2** [reply]
> Is that a good "hm" or a bad one ?

It was a "hm" hm))). In my opinion, it is not so neccessary to add this function, but who am I to resist the will of community? 🐱

**Comment by farag2** [comment]
Added in the last commit: https://github.com/farag2/Sophia-Script-for-Windows/commit/2e089c404a3c7b66ebafdbebb72c7dd4b097838e
